fe/fe-core/src/main/java/org/apache/doris/planner/DistributedPlanner.java

```diff
@@ -97,11 +97,18 @@ public DistributedPlanner(PlannerContext ctx) {
             isPartitioned = true;
         }
         long perNodeMemLimit = ctx_.getQueryOptions().mem_limit;
+        double autoBroadcastJoinThresholdPercentage = ctx_.getQueryOptions().getAutoBroadcastJoinThreshold();
+        if (autoBroadcastJoinThresholdPercentage > 1) {
+            autoBroadcastJoinThresholdPercentage = 1.0;
+        } else if (autoBroadcastJoinThresholdPercentage <= 0) {
+            autoBroadcastJoinThresholdPercentage = -1.0;
+        }
+        long autoBroadcastJoinThreshold = (long)(perNodeMemLimit * autoBroadcastJoinThresholdPercentage);
         if (LOG.isDebugEnabled()) {
             LOG.debug("create plan fragments");
-            LOG.debug("memlimit=" + Long.toString(perNodeMemLimit));
+            LOG.debug("auto broadcast threshold = " + autoBroadcastJoinThreshold);
         }
-        createPlanFragments(singleNodePlan, isPartitioned, perNodeMemLimit, fragments);
+        createPlanFragments(singleNodePlan, isPartitioned, autoBroadcastJoinThreshold, fragments);
```

Comment:
Also, the input params 'mem_limit' of function 'createPlanFragments' is also not required

Reply:
fix it later